src/Pattern/types.ts

```diff
@@ -13,5 +13,6 @@ type ExportedKeys =
 export type PatternOptions = Partial<Pick<Pattern, ExportedKeys>>;
 
 export type SerializedPatternOptions = PatternOptions & {
+  type: 'pattern';
```

Comment:
i m not sure why on toObject we still emit 'pattern' instead of 'Pattern', but that's it.